crowbar_framework/app/models/service_object.rb

```diff
@@ -919,6 +919,35 @@ def self.proposal_to_role(proposal, bc_name)
     RoleObject.new role
   end
 
+  #  we can speed-up the application of (n+1)th role if both(n,n+1)
+  #  roles are being applied on the same node.
+  #
+  #  eg. In our 2node deployment ceilometer{server,central} are always
+  #  applied on the same node, given that they have different priorities,
+  #  they are to be applied, one after the other.
+  #
+  #  In other words: it's actually reducing the number of times chef-client
+  #  is run rather than speeding up execution of any single run, by
+  #  merging batches together
+  #
+  #  a batch is [roles, nodes]
```

Comment:
Thanks - I find comments like this really helpful for understanding the data structures being manipulated.